theseus/embodied/kinematics/kinematics_model.py

```diff
@@ -47,25 +47,35 @@ class UrdfRobotModel(KinematicsModel):
     def __init__(self, urdf_path: str):
         self.drm_model = drm.DifferentiableRobotModel(urdf_path)
 
+    def _postprocess_quaternion(self, quat):
+        # Convert quaternion convention (DRM uses xyzw, Theseus uses wxyz)
+        quat1 = torch.cat([quat[..., 3:], quat[..., :3]])
+
+        # Normalize quaternions
+        quat2 = quat1 / torch.linalg.norm(quat1)
+
+        return quat2
+
     def forward_kinematics(self, joint_states: RobotModelInput) -> Dict[str, LieGroup]:
         """Computes forward kinematics
         Args:
             joint_states: Vector of all joint angles
         Outputs:
-             Dictionary that maps link name to link pose
-
-        (Note: differentiable robot model uses the xyzw quaternion convention,
-        while theseus uses wxyz, hence the conversion.)
+            Dictionary that maps link name to link pose
         """
+        # Check input dimensions
+        assert joint_states.shape[-1] == len(self.drm_model.get_joint_limits())
```

Comment:
We will have to convert the input to torch.Tensor to query DRM anyways, it feels weird to be converting an input torch.Tensor to th.Vector, then converting it back again.
Also the current check only checks the dimension of the joint space, instead of checking for batch dims.

Reply:
Ah makes sense then to not convert back and forth. In that case does it make sense to check for batch dim or is that handled inside DRM? Also instead of just assert probably raise an error with a message.

Reply:
Batch dim is checked and converted within DRM :)